src/soc/ibm/power9/tor.c

```diff
@@ -0,0 +1,769 @@
+/* SPDX-License-Identifier: GPL-2.0-only */
+
+#include "tor.h"
+
+#include <cpu/power/mvpd.h>
+
+#include <assert.h>
+#include <endian.h>
+#include <string.h>
+
+#include "rs4.h"
+
+#define UNDEFINED_PPE_TYPE (uint8_t)0xff
+
+#define NUM_OF_CORES   (uint8_t)24
+#define NUM_OF_QUADS   (uint8_t)6
+#define CORES_PER_QUAD (NUM_OF_CORES/NUM_OF_QUADS)
```

Comment:
These will have to be moved to some higher level header, but I think it can be left for later.